cmake/stm32/devices.cmake

```diff
@@ -1116,14 +1116,68 @@ set(STM32_ALL_DEVICES
     WLE5JC
 )
 
-function(stm32_get_devices_by_family DEVICES)
+# Store a list of devices into a given STM_DEVICES list.
+# You can also specify multiple device families. Examples:
+# Get list of all devices for H7 family: stm32_get_devices_by_family(STM_DEVICES FAMILY H7)
+# Get list of all devices: stm32_get_devices_by_family(STM_DEVICES)
+function(stm32_get_devices_by_family STM_DEVICES)
+    # Specify keywords for argument parsing here
     set(ARG_OPTIONS "")
-    set(ARG_SINGLE FAMILY)
-    set(ARG_MULTIPLE "")
+    set(ARG_SINGLE "")
+    set(ARG_MULTIPLE FAMILY)
+
+    if(ARGC GREATER_EQUAL 1)
+        if(NOT ${ARGV0} STREQUAL "STM_DEVICES")
+            message(WARNING "Passed list ${ARGV0} invalid. Still setting STM_DEVICES variable")
+        endif()
+    endif()
+    if(ARGC MATCHES 2)
+        if(NOT ARGV1 STREQUAL "FAMILY")
+            message(WARNING "Unknown keyword: ${ARGV1}. Matching all devices..")
+        endif()
```

Comment:
I think this can be handled by `cmake_parse_arguments` see https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html
`<prefix>_KEYWORDS_MISSING_VALUES` and `<prefix>_UNPARSED_ARGUMENTS`
I think it will be safer and avoid unexpected corner cases